### PR TITLE
Change error message for default export in namespace

### LINKED ISSUE #11092: Bad error message for default export in namespace
``` ts
namespace Foo {
  export default foo;
}
```

Actual: "An export assignment cannot be used in a namespace."
Expected: "A default export can only be used in an ECMAScript-style module."

### PR DESCRIPTION
**Pull request by YuichiNukiyama**
Fixes #11092

### CODE REVIEW

**Comment by YuichiNukiyama** [comment]
ping @DanielRosenwasser

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -6507,6 +6507,9 @@ namespace ts {
                     else if (source.symbol && source.flags & TypeFlags.ObjectType && globalObjectType === source) {
                         reportError(Diagnostics.The_Object_type_is_assignable_to_very_few_other_types_Did_you_mean_to_use_the_any_type_instead);
                     }
+                    else if (source.symbol && source.flags & TypeFlags.ObjectType && globalObjectType === source) {
```
Comment (RyanCavanaugh) [inline src/compiler/checker.ts]: Needs remerge?

File: tests/baselines/reference/parserExportAssignment5.errors.txt [hunk 2]
```diff
@@ -1,9 +1,9 @@
-tests/cases/conformance/parser/ecmascript5/ExportAssignments/parserExportAssignment5.ts(2,5): error TS1063: An export assignment cannot be used in a namespace.
+tests/cases/conformance/parser/ecmascript5/ExportAssignments/parserExportAssignment5.ts(2,5): error TS1063: A default export can only be used in an ECMAScript-style module.
 
 
 ==== tests/cases/conformance/parser/ecmascript5/ExportAssignments/parserExportAssignment5.ts (1 errors) ====
     module M {
         export = A;
         ~~~~~~~~~~~
-!!! error TS1063: An export assignment cannot be used in a namespace.
+!!! error TS1063: A default export can only be used in an ECMAScript-style module.
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/parserExportAssignment5.errors.txt]: This is a regression - the old error is actually more appropriate here.

**Comment by DanielRosenwasser** [comment]
@YuichiNukiyama would you still be interested in fixing up the bug? I think all you'd need to do is still issue the old error message for `export =`, but use the new one for `export default`

**Comment by YuichiNukiyama** [comment]
Sorry for the delay. How is this?

**Comment by DanielRosenwasser** [comment]
Looks good! However, it looks like you've added a test from one of your older PRs (`assigningFromObjecToAnythingElse`). If you delete that I think we can pull this change in.

**Comment by DanielRosenwasser** [comment]
@YuichiNukiyama `assigningFromObjecToAnythingElse.errors.txt` is still there

**Comment by DanielRosenwasser** [comment]
`assigningFromObjecToAnythingElse.ts` should still be removed and `asiPublicPrivateProtected.ts` should be restored.

**Comment by YuichiNukiyama** [comment]
@DanielRosenwasser fix it.

**Pull request merged**